packages/worker/src/api/controllers/global/users.ts

```diff
@@ -380,51 +383,60 @@ export const inviteAccept = async (
 ) => {
   const { inviteCode, password, firstName, lastName } = ctx.request.body
   try {
-    // info is an extension of the user object that was stored by global
-    const { email, info }: any = await cache.invite.getCode(inviteCode)
-    await cache.invite.deleteCode(inviteCode)
-    const user = await tenancy.doInTenant(info.tenantId, async () => {
-      let request: any = {
-        firstName,
-        lastName,
-        password,
-        email,
-        admin: { global: info?.admin?.global || false },
-        roles: info.apps,
-        tenantId: info.tenantId,
-      }
-      let builder: { global: boolean; apps?: string[] } = {
-        global: info?.builder?.global || false,
-      }
+    await locks.doWithLock(
+      {
+        type: LockType.AUTO_EXTEND,
+        name: LockName.PROCESS_USER_INVITE,
+        resource: inviteCode,
+        systemLock: true,
+      },
+      async () => {
+        // info is an extension of the user object that was stored by global
+        const { email, info } = await cache.invite.getCode(inviteCode)
```

Comment:
Do we also want to do a check here to make sure a user with this email doesn't already exist?